### ISSUE TITLE
Symlinks executable name on onefile binaries on linux

### ISSUE THREAD

**Issue by DavidVentura** [pyinstaller/pyinstaller#3823]
Hello

When running a `onefile` binary on linux the process name of the [second process](https://pyinstaller.readthedocs.io/en/v3.3.1/advanced-topics.html) is showing the name of the proper binary instead of the symlink. 

Binary: `~/binary123`
Symlink: `~/symlink123`

```
$ grep Name /proc/$PARENT_PID/status
symlink123
```

```
$ grep Name /proc/$CHILD_PID/status
binary123
```

both `/proc/$CHILD_PID/cmdline` and `/proc/$PARENT_PID/cmdline` are equal (and show symlink123)

I would expect the CHILD process to show also the symlink as its name. 

I see [here](https://pyinstaller.readthedocs.io/en/v3.3.1/runtime-information.html) some information about this; but this behavior is not consistent with normal linux symlinks/binaries (which show the `basename` of the symlink in /proc/self/status) and this causes telegraf to misbehave/misreport

**Comment by htgoebel** [comment]
Please provide a pull-request if you want this to be fixed.

**Comment by DavidVentura** [comment]
By overriding `archive_status->argv[0]` on `pyi_main.c` (line 103) I can get `/proc/$CHILD/cmdline` to change; but does not modify what is reported as `name` on `/proc/$CHILD/status`

I think the problem comes from `execvp(thisfile, argv)` on `pyi_utils.c`; a quick debug shows that `thisfile` points the the resolved (!) symlink and not to the original file which started the parent process.

I will try to fix it now

**Comment by DavidVentura** [comment]
Ok, then the problem resides in `pyi_path.c`
```
    result = readlink("/proc/self/exe", execfile, PATH_MAX);  /* Linux */
```

this will resolve all symlinks (as expected from the docs).

If I were to ignore this; the fallback is `realpath(rel, abs)` (where `rel` here is the symlink name) which also resolves symlinks..

What I did to get this working was 

```c
    // appname is argv[0] -> ../test/symlink
    pyi_path_basename(basename, appname);
    // basename == "symlink"
    pyi_path_dirname(dirname, appname);
    // dirname == "../test"
    crap = realpath(dirname, full_dirname);
    // full_dirname == "/home/david/test"
    // i needed to store the result somewhere to make the compiler happy
    result = (pyi_path_join(execfile, full_dirname, basename) == NULL);
    // result points to execfile
    // execfile == "/home/david/test/symlink"
    result = strlen(execfile);
    // I am assigning this to result as the code that follows will put a null-terminator on *(execfile+result)
    // even though this is null terminated already
```


```

**Comment by bwoodsend** [comment]
Fixed in #3829.
